contracts/dtoken/src/interest_rate_model.rs

```diff
@@ -56,6 +73,28 @@ impl InterestRateModel{
     pub fn set_reserve_factor(&mut self, value: Ratio){
         self.reserve_factor = value;
     }
+
+    #[private]
+    pub fn set_supply_block_by_user(&mut self, account: AccountId, block_height: BlockHeight){
+        self.account_supply_block.insert(&account, &block_height);
+    }
+
+    #[private]
+    pub fn set_supply_interest_by_user(&mut self, account: AccountId, interest: Ratio){
+        self.account_supply_interest.insert(&account, &interest);
+    }
+
+    pub fn get_accrued_supply_interest(&mut self, account: AccountId, supply_rate: Ratio) -> Ratio {
+        let current_block_height = block_height();
+        if current_block_height == self.get_supply_block_by_user(account.clone()){

```

Comment:
@dariacherniak we should accumulate user accured interest instead of rate (because rate applies to amount which may change)

this function should calculate how much accrued interest user get since last block and accumulate somewhere (lately we will introduce function that will allow user to claim those accumulated interest) like this
```
let accrued_interest = user_supplies * self.get_supply_rate(..) * (current_block - last_accrued_block);
self.set_accrued_interest(user, prev_interest + accrued_interest);
```

also we need to call this function everywhere we updates supplies (for now it's only supply \ withdraw methods)